src/chain/optimism.rs

```diff
@@ -0,0 +1,286 @@
+        OpTxType::Legacy | OpTxType::Eip2930 => tx
+            .inner
+            .gas_price
+            .map(U256::from)
+            .ok_or(OptimismTransactionParsingError::MissingGasPrice),
+        OpTxType::Eip1559 | OpTxType::Eip4844 => tx
+            .inner
+            .max_fee_per_gas
+            .map(U256::from)
+            .ok_or(OptimismTransactionParsingError::MissingMaxFeePerGas),
+        OpTxType::Deposit => Ok(U256::ZERO),
+    }
+}
+
+/// Convert [Transaction] to [OptimismFields]
+pub(crate) fn get_optimism_fields(
+    tx: &Transaction,
+) -> Result<OptimismFields, OptimismTransactionParsingError> {
+    let envelope_buf = {
+        let tx_type = tx.inner.transaction_type.unwrap_or_default();
+        let op_tx_type = OpTxType::try_from(tx_type)
+            .map_err(|_err| OptimismTransactionParsingError::UnexpectedType(tx_type))?;
+        let inner = tx.inner.clone();
+        let tx_envelope = match op_tx_type {
+            OpTxType::Legacy => Signed::<TxLegacy>::try_from(inner).map(OpTxEnvelope::from),
+            OpTxType::Eip2930 => Signed::<TxEip2930>::try_from(inner).map(OpTxEnvelope::from),
+            OpTxType::Eip1559 => Signed::<TxEip1559>::try_from(inner).map(OpTxEnvelope::from),
+            OpTxType::Eip4844 => Signed::<TxEip4844>::try_from(inner).map(OpTxEnvelope::from),
+            OpTxType::Deposit => {
+                let tx_deposit = TxDeposit {
+                    source_hash: tx
+                        .source_hash
+                        .ok_or(OptimismTransactionParsingError::MissingSourceHash)?,
+                    from: tx.inner.from,
+                    to: tx.inner.to.into(),
+                    mint: tx.mint,
+                    value: tx.inner.value,
+                    gas_limit: tx.inner.gas,
+                    is_system_transaction: tx.is_system_tx.unwrap_or_default(),
+                    input: tx.inner.input.clone(),
+                };
+                Ok(OpTxEnvelope::from(tx_deposit))
+            }
+        }
+        .map_err(|err| OptimismTransactionParsingError::ConversionError(err.to_string()))?;
+
+        let mut envelope_buf = Vec::<u8>::new();
+        tx_envelope.encode_2718(&mut envelope_buf);
+        Bytes::from(envelope_buf)
+    };
+
+    Ok(OptimismFields {
+        source_hash: tx.source_hash,
+        mint: tx.mint,
+        is_system_transaction: tx.is_system_tx,
+        enveloped_tx: Some(envelope_buf),
+    })
+}
+
+impl PevmChain for PevmOptimism {
+    type Transaction = Transaction;
+    type BlockSpecError = OptimismBlockSpecError;
+    type TransactionParsingError = OptimismTransactionParsingError;
+
+    fn id(&self) -> u64 {
+        self.id
+    }
+
+    fn build_tx_from_alloy_tx(&self, tx: alloy_rpc_types::Transaction) -> Self::Transaction {
+        Transaction {
+            inner: tx,
+            ..Transaction::default()
+        }
+    }
+
+    fn get_block_spec(&self, header: &Header) -> Result<SpecId, Self::BlockSpecError> {
```

Comment:
Let's add a comment to support RISE (we'll do it soon) and other OP testnets (community contributions).

Reply:
I added. Feel free to suggest a change if you want.